helix-term/src/commands.rs

```diff
@@ -2973,6 +3037,34 @@ pub mod cmd {
             fun: force_buffer_close,
           completer: Some(completers::buffer),
         },
+        TypableCommand {
+            name: "buffer-close-others",
+            aliases: &["bco", "bcloseother"],
+            doc: "Close all buffers but the currently focused one.",
+            fun: buffer_close_others,
+            completer: None,
+        },
+        TypableCommand {
+            name: "buffer-close-others!",
+            aliases: &["bco!", "bcloseother!"],
+            doc: "Close all buffers but the currently focused one.",
+            fun: force_buffer_close_others,
+            completer: None,
+        },
+        TypableCommand {
+            name: "buffer-close-all",
+            aliases: &["bca", "bcloseall"],
+            doc: "Close all buffers, without quiting.",
+            fun: buffer_close_all,
+            completer: None,
+        },
+        TypableCommand {
+            name: "buffer-close-all!",
+            aliases: &["bca!", "bcloseall!"],
+            doc: "Close all buffers forcefully (ignoring unsaved changes), without quiting.",
+            fun: force_buffer_close_all,
+            completer: None,
+        },
```

Comment:
It shouldn't be possible to have no documents left because the editor will end up closing (no view pointing to anything). You'd have to close all the buffers then open a new scratch buffer.

Reply:
Something else in the code handles this automatically! I was surprised I didn't have to handle this case as well.

Reply:
To be clear, when you utilize the buffer-close-all command, it will in fact remove all currently loaded buffers (closing all but the last view), and spawn a new scratch buffer to keep the last view alive. This is a handy thing to have if you want to clear out the buffers you're working on without losing the loaded LSP instance, for example.